src/Jackalope/Version/VersionHistory.php

```diff
@@ -214,9 +231,56 @@ public function hasVersionLabel($label, $version = null)
      */
     public function getVersionLabels($version = null)
     {
-        // @codeCoverageIgnoreStart
-        throw new NotImplementedException();
-        // @codeCoverageIgnoreEnd
+        $this->initVersionLabels();
+        if ($version === null) {
+            return array_keys($this->versionLabels);
+        }
+
+        $versions = $this->getAllVersions();
+        $versionIsInHistory = false;
+
+        foreach ($versions as $versionCheck) {
+            /* @var VersionInterface $versionCheck */
+            if ($versionCheck->getIdentifier() == $version->getIdentifier()) {
+                $versionIsInHistory = true;
+                break;
+            }
+        }
+
+        if (!$versionIsInHistory) {
+            throw new VersionException(sprintf('Version %s not found in history of %s', $version->getIdentifier(), $this->getPath()));
+        }
+
+        $result = array();
+        foreach ($this->versionLabels as $label => $labelVersion) {
+            /* @var VersionInterface $labelVersion */
+            if ($labelVersion->getIdentifier() == $version->getIdentifier()) {
+                $result[] = $label;
+            }
+        }
+
+        return $result;
+    }
+
+    /**
+     * This method fetches all version labels, if the cache array is not initialized yet.
+     */
+    private function initVersionLabels()
+    {
+        if (!is_null($this->versionLabels)) {
+            return;
+        }
+
+        $this->versionLabels = array();
+        $node = $this->getNode('jcr:versionLabels');
+        foreach ($node->getProperties() as $property) {
+            /* @var Property $property */
+            if ($property->getName()!= "jcr:primaryType") {
```

Comment:
should we use PHPCR\Util\NodeHelper::isSystemItem? it will basically check whether the name prefix is `jcr:`.